controller/attribute/manager.py

```diff
@@ -251,17 +252,28 @@ def __calculate_user_attribute_all_records(
         return
     util.add_log_to_attribute_logs(project_id, attribute_id, "Finished writing.")
 
-    if attribute.get(project_id, attribute_id).data_type == DataTypes.TEXT.value:
+    attribute_item = attribute.get(project_id, attribute_id)
+    if attribute_item.data_type == DataTypes.TEXT.value:
         util.add_log_to_attribute_logs(
             project_id, attribute_id, "Triggering tokenization."
         )
-        tokenization.delete_docbins(project_id, with_commit=True)
-        tokenization.delete_token_statistics_for_project(project_id, with_commit=True)
-
-        while record.count_tokenized_records(project_id) > 0:
-            time.sleep(2)
+        try:
+            request_tokenize_calculated_attribute(
+                project_id, user_id, attribute_item.id
+            )
+        except:
+            record.delete_user_created_attribute(
+                project_id=project_id,
+                attribute_id=attribute_id,
+                with_commit=True,
+            )
```

Comment:
not sure if we should simply delete the attribute if the tokenization fails maybe set the attribute to error?